### PR TITLE
Hibernatize SampleQC

### PR DESCRIPTION
**Pull request by callunity**

### CODE REVIEW

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/AbstractSampleQC.java [hunk 1]
```diff
@@ -31,10 +36,14 @@
...
  * @since 0.0.2
  */
+@MappedSuperclass
 public abstract class AbstractSampleQC extends AbstractQC implements SampleQC {
   public static final String UNITS = "ng/&#181;l";
 
   private Double results;
+
+  @ManyToOne(targetEntity = SampleImpl.class)
+  @PrimaryKeyJoinColumn
```
Comment (djcooke) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/AbstractSampleQC.java]: This is only for 1:1 relationships, when both tables share the same primary key

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/AbstractSample.java [hunk 2]
```diff
@@ -80,17 +83,20 @@
...
+  private final Collection<Experiment> experiments = new HashSet<>();
 
   @Transient
   @JsonManagedReference
-  private final Collection<Library> libraries = new HashSet<Library>();
+  private final Collection<Library> libraries = new HashSet<>();
 
-  @Transient
-  private Collection<SampleQC> sampleQCs = new TreeSet<SampleQC>();
+  @OneToMany(targetEntity = SampleQCImpl.class)
```
Comment (apmasell) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/AbstractSample.java]: Doesn't this need a `@JoinColumn`?

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/AbstractSample.java [hunk 3]
```diff
@@ -80,17 +83,20 @@
   private Project project;
 
   @Transient
-  private final Collection<Experiment> experiments = new HashSet<Experiment>();
+  private final Collection<Experiment> experiments = new HashSet<>();
 
   @Transient
```
Comment (apmasell) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/AbstractSample.java]: Isn't this loadable by `@OneToMany`?

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/AbstractSample.java (hunk 2 above)
Comment (callunity) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/AbstractSample.java]: No, but it does need a `mappedBy`

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/AbstractSampleQC.java [hunk 4]
```diff
@@ -43,7 +43,7 @@
   private Double results;
 
   @ManyToOne(targetEntity = SampleImpl.class)
-  @PrimaryKeyJoinColumn
+  @Column(name = "sample_sampleId")
```
Comment (djcooke) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/AbstractSampleQC.java]: Should be `@JoinColumn`

Comment (djcooke) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/AbstractSampleQC.java]: Disregard - was already fixed

**Pull request merged**